### PR TITLE
EVEREST-1292 | Migrate backup-storages and monitoring-configs during 1.2.0 upgrade

### PR DESCRIPTION
**Pull request by mayankshah1607**

### CODE REVIEW

**Review by oksana-grishchenko — APPROVED** [review]
LGTM. I would've concerned about the code duplication but it's a one-time thing, I don't think we need to overcomplicate (it looks like making the duplicated functions reusable would require even more code).

File: pkg/upgrade/upgrade.go [hunk 1]
```diff
@@ -215,12 +215,16 @@ func (u *Upgrade) Run(ctx context.Context) error {
 					return err
 				}
 			}
-			// RBAC is added in 1.2.x, so if we're upgrading to that version, we need to ensure
-			// that the RBAC configmap is present.
 			if common.CheckConstraint(upgradeEverestTo, "~> 1.2.0") {
```
Comment (recharte) [inline pkg/upgrade/upgrade.go:218]: What if we eventually release a `1.2.1`? Can we make this check a `1.2.x`?

Comment (mayankshah1607) [reply pkg/upgrade/upgrade.go:218]: Yes, that constraint does indeed check for `1.2.x`. The `~>` is called the [pessimistic constraint operator](https://developer.hashicorp.com/terraform/language/expressions/version-constraints#version-constraint-syntax)

Comment (recharte) [reply pkg/upgrade/upgrade.go:218]: Ohhh yeah, sorry for that I didn't properly check it :see_no_evil:

**Pull request merged**